rules.go

```diff
@@ -7,45 +7,82 @@ import (
 
 type NewRuleNoJson NewRule
 
+// AuthenticationMode is an enum of authentication modes used by pulsar rules.
```

Comment:
Pulsar should be capitalised, based on how Apache uses the name in their own docs.

Reply:
i.e. https://pulsar.apache.org/docs/next/concepts-overview/